% Template for PLoS
% Version 1.0 January 2009
%
% To compile to pdf, run:
% latex plos.template
% bibtex plos.template
% latex plos.template
% latex plos.template
% dvipdf plos.template

\documentclass[10pt]{article}

% amsmath package, useful for mathematical formulas
\usepackage{amsmath}
% amssymb package, useful for mathematical symbols
\usepackage{amssymb}

% graphicx package, useful for including eps and pdf graphics
% include graphics with the command \includegraphics
\usepackage{graphicx}

% cite package, to clean up citations in the main text. Do not remove.
\usepackage{cite}

\usepackage{color} 

% Use doublespacing - comment out for single spacing
%\usepackage{setspace} 
%\doublespacing

\usepackage[frenchb,english]{babel}
\usepackage{bm}
\usepackage{cite}
\usepackage{amsthm}
\usepackage{amsfonts}
\usepackage{ccaption}
\usepackage{url}

% Text layout
\topmargin 0.0cm
\oddsidemargin 0.5cm
\evensidemargin 0.5cm
\textwidth 16cm 
\textheight 21cm

% Bold the 'Figure #' in the caption and separate it with a period
% Captions will be left justified
\usepackage[labelfont=bf,labelsep=period,justification=raggedright]{caption}

% Use the PLoS provided bibtex style
\bibliographystyle{plos2009}

% Remove brackets from numbering in List of References
\makeatletter
\renewcommand{\@biblabel}[1]{\quad#1.}
\makeatother

\author{In alphabetic order: \\Simon Cauchemez, Anne Cori, Neil Ferguson, Christophe Fraser, Thibaut Jombart}
\title{A model of MRSA transmission incorporating epidemiological and genetic data}

\begin{document}

% Title must be 150 words or less
% \begin{flushleft}
\maketitle

\section*{Observed data ($Y$)}

For each patient $i=1,\ldots,N$ admitted to one of the wards in the study period, we denote 
\begin{itemize}
	\item $w_i$ the ward where the patient is admitted ($1$ for adult ICU, $2$ for paediatric ICU)
	\item $k_i$ the number of times the patient is admitted (1 if no readmission)
	\item $A_i$ and $D_i$ vectors containing the times of admission and discharge from the ward
	\item $P_i$ and $N_i$ vectors containing the times of positive and negative swabs (positive defined as any of the samples taken is positive ; negative defined as all samples taken are negative).
	\item $p_i$ and $n_i$ the size of those vectors, ie the number of positive and negative swabs.
 	\item $S_i = \{s_i^1, \ldots, s_i^{m_i}\}$ a set of $m_i$ genetic sequences of MRSA isolated in patient $i$ at times 
$T_i = \{t_i^1, \ldots, t_i^{m_i}\}$; collection dates $T_i$ are ordered so that $t_i^k \leq t_i^{k+1}$.
% 	\item $s_i$ a genetic sequence of the MRSA isolated in patient $i$ at time $t_i$.
% 	\item $\delta_{ab}$ and $\gamma_{ab}$ the number of transitions and transversions between isolates $a$ and $b$.
% 	\item $L_{ab}$ the length of the DNA sequence comparable between $a$ and $b$.
 	\item $d_{s_i^k,s_j^q}$ the number of transitions between sequence $k$ of patient $i$ and sequence $q$ of patient $j$.
	\item $g_{s_i^k,s_j^q}$ the number of transversions between sequence $k$ of patient $i$ and sequence $q$ of patient $j$.
	\item $l_{s_i^k,s_j^q}$ the number of typed nucleotides common to sequences $s_i^k$ and $s_j^q$.
\end{itemize}

\section*{Augmented (unobserved) data ($Z$)}

For each patient $i$ admitted to one of the wards in the study period, we denote 
\begin{itemize}
	\item $C_i$ the colonisation time (we assume no supercolonisations)
	\item $E_i$ the time of end of colonisation.
\end{itemize}

% For each pair of sequences $s_i^k$ and $s_j^q$ in patients $i$ and $j$, we denote
% \begin{itemize}
% % 	\item $\tau_{s_i^k,s_j^q}$ the time to the most recent common ancestor (tMRCA) (before the earliest collection date).
% 	\item $b_{s_i^k,j}$ an indicator function with value 1 if a direct ancestor to sequence $s_i^k$ has been sampled in patients $i$ or $j$, and 0 otherwise.
% \end{itemize}


\noindent We denote $I_w\left(t\right)=\sum_{i=1}^{N} \mathbf{1}_{\left\lbrace w_i=w \right\rbrace} \mathbf{1}_{\left\lbrace C_i \leq t < E_i \right\rbrace} \sum_{l=1}^{k_i} \mathbf{1}_{\left\lbrace A_i[l] \leq t < D_i[l] \right\rbrace} $  the number of patients in ward $w$ who are colonized at time $t$.

\section*{Parameters ($\theta$)}

Parameters of the model are: 
\begin{itemize}
	\item $\beta$ a 2 by 2 matrix containing $\beta_{i \leftarrow j}$, the person to person transmission rate from ward $j$ to ward $i$
	\item $\beta_{\text{ward} \leftarrow \text{out}}$ the force of infection from outside the 2 wards applied to patients in the wards
	\item $\beta_{\text{out} \leftarrow \text{out}}$ the force of infection from outside the 2 wards applied to patients when they are not in the wards (eg inbetween two admissions)
	\item $Sp$ the specificity of the testing, ie the probability of getting a negative test given uncolonized (assumed $100\%$)
	\item $Se$ the sensitivity of the testing, ie the probability of getting a positive test given colonized
	\item $\pi$ the probability of being already colonized at first admission
	\item $\mu$ and $\sigma$ the mean and standard deviation of the duration of colonization.
	\item $\nu_1$ and $\nu_2$ the rate of transitions ($A \leftrightarrow G$ and $C \leftrightarrow T$) and transversions (other changes) of the DNA sequences. In practice, we will use $\nu_2=\kappa\nu_1$ with $\kappa \in \mathbb{R}_+$.
% 	\item $\tau$ the time to the most recent common ancestor (MRCA) of a pair of isolates for indirect ancestries (before the earliest collection date).
% 	\item $\alpha$ the probability that two sampled isolates belong to the same lineage (i.e., one is the direct ancestor of the other).
	\item $\alpha_i$ is the within host 'pathogenic diversity', defined as the number of pathogenic lineages infecting patient $i$; all lineages are supposed to be as likely to be have been sequenced.
\end{itemize}

\section*{Statistical Model}

In the following, $\mathbf{1}_{\left\lbrace.\right\rbrace}$ denotes the indicator function, defined by $\mathbf{1}_{\left\lbrace X \right\rbrace}=1$ if $X$ is true, and $0$ otherwise.

The joint density of the observed data, the augmented data, and the model parameters is: 

\begin{equation} \label {JointDensity}
\begin{split}
P\left(\bm{Y},\bm{Z},\bm{\theta}\right) = P\left(\bm{Y}|\bm{Z}\right)P\left(\bm{Z}|\bm{\theta}\right)P\left(\bm{\theta}\right)\\ \nonumber
\end{split}
\end{equation}

\noindent where $P\left(\bm{Y}|\bm{Z}\right)$, $P\left(\bm{Z}|\bm{\theta}\right)$ 
and $P\left(\bm{\theta}\right)$ refer to the observation level, the transmission level
 and the prior level respectively.

%%%%%%%%

\subsection*{Observation level}

%%%%%%%%

The observation level ensures that the observed data are consistent with the augmented data:

\begin{eqnarray*}
P\left(\bm{Y}|\bm{Z}\right) = &
\bm{\prod}_{i=1}^N & % product on individuals
 \mathbf{1}_{\left\lbrace C_i < E_i \right\rbrace} \\
 &&
 \bm{\prod}_{j=1}^{p_i} % positive tests
\left(
\left( \mathbf{1}_{\left\lbrace P_i[j] < C_i \right\rbrace}+\mathbf{1}_{\left\lbrace P_i[j] \geq E_i \right\rbrace} \right) \times \left(1-Sp\right) % false positive
+ \mathbf{1}_{\left\lbrace C_i \leq P_i[j] < E_i \right\rbrace} \times Se % true positive
\right) \\ 
& & \bm{\prod}_{k=1}^{n_i} % negative tests
\left(
\left( \mathbf{1}_{\left\lbrace N_i[k] < C_i \right\rbrace}+\mathbf{1}_{\left\lbrace N_i[k] \geq E_i \right\rbrace} \right) \times Sp % true negative
+ \mathbf{1}_{\left\lbrace C_i \leq N_i[k] < E_i \right\rbrace} \times \left(1-Se\right) % false negative
\right) \\ 
\end{eqnarray*}

The first line describes the positive tests, which can be either false positives (first term) or true positives (second term). 
The second line describes the negative tests, which can be either true negatives (first term) or false negatives (second term). 

%%%%%%%%

\subsection*{Transmission level (discrete time version ; time step = half day or day ?)} 

%%%%%%%%

In the discrete version $A_i[k]$ is the first time step where individual $i$ is in hospital (for his/her $k_{th}$ stay), and $D_i[k]$ is the first time step where he/she is out of hospital (after his/her $k_{th}$ stay). Individual $i$ can transmit staph aureus from time step $C_i$ to time step $E_i-1$. 

\begin{eqnarray*}
P\left(\bm{Z}|\bm{\theta}\right) = 
\bm{\prod}_{i=1}^N % product on individuals
\left( \Omega_i^{\left(1\right)} + \Omega_i^{\left(2\right)} \right) \times \left(\Phi_{\mu,\sigma}\left(E_i-C_i+0.5\right) - \Phi_{\mu,\sigma}\left(E_i-C_i-0.5\right) \right)
\end{eqnarray*}

\noindent where $\Phi_{\mu,\sigma}$ is the cumulative density function of a Gamma distribution with mean $\mu$ and standard deviation $\sigma$ (we assume that the duration of colonisation is Gamma distributed), and: 

\begin{eqnarray*}
\Omega_i^{\left(1\right)} & = & \pi \times \mathbf{1}_{\left\lbrace C_i < A_i[1] \right\rbrace}  \\
\Omega_i^{\left(2\right)}  & = & \left( 1-\pi \right) \times \mathbf{1}_{\left\lbrace C_i \geq A_i[1] \right\rbrace} \times e^{-\sum_{t=A_i[1]}^{C_i-1}\lambda_i\left(t\right)} \left( 1 - e^{-\lambda_i\left(C_i\right)} \right) \eta_i\left(C_i\right)\\
\end{eqnarray*}
\bigskip

$\Omega_i^{\left(1\right)}$ is the probability that individual $i$ is colonized before his/her first admission in the wards ; $\Omega_i^{\left(2\right)}$ is the probability that individual $i$ is colonized after his/her first admission in the wards. 
$\lambda_i\left(t\right)$ is the force of transmission applied to individual $i$ at time $t$. It is equal to: 
\begin{eqnarray*}
\lambda_i^{\left(t\right)} & = & \sum_{w=1}^{2} \beta_{w_i \leftarrow w} I_w\left(t\right) + \beta_{\text{ward} \leftarrow \text{out}} \text{ if individual $i$ is in a ward at time $t$}  \\
 & = & \beta_{\text{out} \leftarrow \text{out}} \text{ otherwise}\\
\end{eqnarray*}
\bigskip
$\eta_i\left(C_i\right)$ is the probability of observing sequences $S_i$ given the possible contamination sources at time $C_i$. It is equal to: 
\begin{eqnarray*}
\eta_i\left(C_i\right) & = & \frac{\sum_{\text{$j$ colonized and in a ward at time $C_i$}} \beta_{w_i \leftarrow w_j} f_{i \leftarrow j} + \beta_{\text{ward} \leftarrow \text{out}}f_{\text{i},\text{ward} \leftarrow \text{out}} }{\sum_{\text{$j$ colonized and in a ward at time $C_i$}} \beta_{w_i \leftarrow w_j} + \beta_{\text{ward} \leftarrow \text{out}} } \text{ if individual $i$ is in a ward at time $C_i$}  \\
 & = & f_{\text{i},\text{out} \leftarrow \text{out}} \text{ otherwise}\\
\end{eqnarray*}
\bigskip

$f_{i,j}$ is the probability of observing sequences $s_i$ and $s_j$ at respective time steps $t_i$ and $t_j$ given that individual $j$ infected individual $i$ (see next section on the genetic likelihood). Similarly, $f_{\text{i},\text{ward} \leftarrow \text{out}}$ and $f_{\text{i},\text{out} \leftarrow \text{out}}$ are the probability of observing sequence $s_i$ at time $t_i$ given that individual $i$ was infected from outside the wards while he was in or outsite hospital respectively. 

%%%%%%%%

\subsection*{Genetic likelihood}

%%%%%%%%

\subsubsection*{Ancestors and lineages}
\noindent We say that $B$ is an \textit{ancestor} of $A$ if and only if there is a path leading from $B$ to $A$ in the directed acyclic graph (DAG) representing the genealogy of $A$.
Put simply, $B$ is an ancestor of $A$ if $A$ derives from $B$.
We will say that $B$ is the \textit{most recent ancestor} (MRA) of $A$ if there is no older ancestor of $A$ in the considered set.
A \textit{lineage} is defined as a set of (temporally ordered) individuals $\{x_1, ..., x_n\}$ so that $x_i$ is the MRA of $x_{i+1}$ for $i=1,\ldots,n-1$.
For instance, in the lineage ($D \rightarrow C \rightarrow B \rightarrow A$), $B,C,D$ are all ancestors of $A$ and $B$ is the MRA of $A$.
The genetic likelihood of $A$ will be defined as the probability of the observed mutations between $B$ and $A$, and is not conditional on previous ancestries.
\\

\subsubsection*{Genetic likelihood of an infection}
The genetic likelihood of the infection of $i$ by $j$ (noted $i \leftarrow j$) relies on how likely it is to observe the genetic differences between sequences in $S_i$ and their most recent ancestors (MRA) in $S_j$.
We first focus on the probability of observing a given sequence $S_i^k$ in $i$ given that $j$ infected $i$.
We will note $\phi(s_i^k, s_j^q)$ the probability that $s_j^q$ is an ancestor of $s_i^k$, defined as:
$$
\phi(s_i^k, s_j^q) = \mathbf{1}_{\{t_i^k \geq t_j^q\}} \times 
\underbrace{\mathcal{P}\left(d_{s_i^k,s_j^q} | \nu_1 (t_i^k - t_j^q) l_{s_i^k,s_j^q}\right)}_{\mbox{transitions}}
\times \underbrace{\mathcal{P}\left(g_{s_i^k,s_j^q} | \nu_2 (t_i^k - t_j^q) l_{s_i^k,s_j^q}\right)}_{\mbox{transversions}}
$$
with:
\begin{itemize}
\item $\mathbf{1}_{\{\mbox{statement}\}}$: indicator function, 1 if 'statement' is true, 0 otherwise
\item $\mathcal{P}(.|\lambda)$: the probability mass function of a Poisson distribution with parameter $\lambda$
\end{itemize}
. The three terms respectively correspond to the indicator function ensuring that $s_j^q$ is older than $s_i^k$, the probability of the observed transitions ($d_{s_i^k,s_j^q}$), and the probability of the observed transversions ($g_{s_i^k,s_j^q}$). 
\\

We are now interested in $\xi(s_i^k, s_j^q)$,the probability that the sequence $s_j^q$ is the MRA of $s_i^k$.
This requires two elements: i) that $s_j^q$ is an ancestor of $s_i^k$, and ii) that no ancestor of $s_i^k$ has been collected after $s_j^q$.
This is given by:
$$
\xi(s_i^k, s_j^q) = \underbrace{\phi(s_i^k, s_j^q)}_{s_j^q \mbox{ ances. of } s_i^k} \times 
\prod_{r=q+1}^{m_j} \underbrace{(1 - \phi(s_i^k, s_j^r))}_{s_j^r \mbox{ not ances. of } s_i^k}
$$


The genetic likelihood also needs to account for the possibility that no ancestor of $s_i^k$ has been isolated and sequenced in $S_j$.
Assuming that all lineages are as likely to have been sequenced, the probability $\gamma(s_i^k, S_j)$ that the sampled $S_j$ contains at least one ancestor of $s_i^k$ is:
$$
\gamma(s_i^k, S_j) = 1 - \mathcal{B}\left(0 \lvert \:\: \sum_{j=1}^{m_j}\mathbf{1}_{\{t_i^k \geq t_j^q\}}, 1/\alpha_i \right)
$$
with:
\begin{itemize}
\item $\mathcal{B}(.\lvert n,p)$: probability mass function of the Binomial distribution with $n$ draws and probability $p$
\item $\sum_{j=1}^{m_j}\mathbf{1}_{\{t_i^k \geq t_j^q\}}$: number of isolates sequenced in patient $j$ and collected before the sequence $s_i^k$
\item $\alpha_i$: number of lineages in patient $j$
\end{itemize}


The probability $p(s_i^k | S_j, i \leftarrow j)$ of observing the sequence $s_i^k$ given that patient $j$ infected patient $i$ can now be computed as:
$$
p(s_i^k | S_j, i \leftarrow j) = 
(\underbrace{\gamma(s_i^k, S_j)}_{\mbox{ances. in }S_j} \times 
\underbrace{\sum_{q=1}^{m_j}\xi(s_i^k, s_j^q)}_{\mbox{prob. MRA for each }S_j} ) + 
\underbrace{1 - \gamma(s_i^k, S_j)}_{\mbox{ances. not sampled}}
$$


The probability of observing the set of sequences $S_i$ given that $j$ infected $i$ is simply computed as the product over all sequences in $S_i$:
\begin{eqnarray*}
p(S_i | S_j, i \leftarrow j) & = & \prod_{k=1}^{m_i} p(s_i^k | S_j, i \leftarrow j) \\
\end{eqnarray*}
For the sake of simplicity, we shall refer to this quantity as $f_{i,j}$.
\\


\subsubsection*{Assumptions of the model}
The genetic model makes a few key assumptions:
\begin{itemize}
 \item different types of mutations happen independently
 \item all lineages within a host are as likely to have been sampled and sequenced; when lineages have different within-host population sizes, this may still be ensured by extensive sequencing where only new haplotypes are retained; this assumption could be relaxed by parametrizing $\alpha_i$ as distributions.
 \item all lineages present in a host are transmitted during a new infection; models excplicitly incorporating possible losses of diversity during the sampling process will be much more complex, and it will likely be difficult to disentangle this from sampling biases.
\end{itemize}


% Note that in the absence of genetic information ($m_i<1$ and $m_j=0$), $b_{s_i^k,j}$ is automatically set to 0.
% The genetic likelihood only makes sense when there are sequences to be compared between $i$ and $j$, that is, when $m_i$ and $m_j$ are greater than 0.
% We cannot simply ommit the genetic term in the global likelihood computation and implicitely assume $p(S_i | S_j, i \leftarrow j) = 1$, as this would give stronger weight to cases without genetic data.
% Data augmentation is not possible either, as the space of possible sequences is huge unless we make very strong assumptions (on the number of clusters and the distributions of distances within and between clusters).

% One practical alternative is to define a weight $g$ for cases with missing DNA information.
% We then define the genetic pseudo-likelihood function $f_{i \leftarrow j} $ as:
% 
% The likelihood incorporating missing DNA informaiton ($m_i m_j =0$) is given by:
% \begin{eqnarray*}
% f_{i \leftarrow j} & = & \mathbf{1}_{\{ m_i m_j > 0 \}} p(S_i | S_j, i \leftarrow j)
% + \mathbf{1}_{\{ m_i m_j = 0 \}}\\
% \end{eqnarray*}

% Different strategies can be used to define $g$.
% The simplest would be fixed weight, $g(x) = cst$.
% Another one is to set $g(x)$ to the average weight of the computable $p(S_i | S_j, i \leftarrow j)$:
% \begin{eqnarray*}
% g(x) = \sum_{i,j,m_i m_j > 0} p(S_i | S_j, i \leftarrow j) / \sum_{i,j} \mathbf{1}_{\{m_i m_j > 0\}}
% \end{eqnarray*}
% Alternatively, we can define $g(x)$ as a given quantile of the distribution of $p(S_i | S_j, i \leftarrow j)$.

% \noindent Alternatively, $g(x)$ could be defined as a function of the priors.



%%%%%%%%

\subsection*{Prior level}

%%%%%%%%

For all model parameters, independent prior distributions were chosen: 
\begin{itemize}
	\item uniform on $[0,1]$ for $Sp$, $Se$, and $\pi$, 
	\item flat exponential (mean 1000) for all other parameters. 
\end{itemize}

  
\section*{Parameter Estimation}

A Markov chain Monte Carlo (MCMC) method was used to sample the joint
posterior distribution $P\left(\bm{Y},\bm{Z},\bm{\theta}\right)$.
$Sp$, $Se$ and $\pi$ were updated using the Gibbs sampler, and all other parameters using a Metropolis algorithm.


\section*{Appendix: Full formulation of the transmission level}

In the discrete version $A_i[k]$ is the first time step where individual $i$ is in hospital (for his/her $k_{th}$ stay), and $D_i[k]$ is the first time step where he/she is out of hospital (after his/her $k_{th}$ stay). Individual $i$ can transmit staph aureus from time step $C_i$ to time step $E_i-1$. 

\begin{eqnarray*}
P\left(\bm{Z}|\bm{\theta}\right) = 
\bm{\prod}_{i=1}^N % product on individuals
\left( \Omega_i^{\left(1\right)} + \Omega_i^{\left(2\right)} + \Omega_i^{\left(3\right)} \right) \times \left(\Phi_{\mu,\sigma}\left(E_i-C_i+0.5\right) - \Phi_{\mu,\sigma}\left(E_i-C_i-0.5\right) \right)
\end{eqnarray*}

\noindent where $\Phi_{\mu,\sigma}$ is the cumulative density function of a Gamma distribution with mean $\mu$ and standard deviation $\sigma$ (we assume that the duration of colonisation is Gamma distributed), and: 

\begin{eqnarray*}
\lefteqn{\Omega_i^{\left(1\right)} = \pi \times \mathbf{1}_{\left\lbrace C_i < A_i[1] \right\rbrace} }\\
\lefteqn{\Omega_i^{\left(2\right)}  = \left( 1-\pi \right) \sum_{l=1}^{k_i} \mathbf{1}_{\left\lbrace A_i[l] \leq C_i < D_i[l] \right\rbrace}}\\
	 && \times \text{exp}\left( - \mathbf{1}_{\left\lbrace l \geq 2 \right\rbrace} \sum_{s=1}^{l-1} \left[ \sum_{t=A_i[s]}^{D_i[s]-1} \left( \sum_{w=1}^{2} \beta_{w_i \leftarrow w} I_w\left(t\right) \right) + \beta_{\text{ward} \leftarrow \text{out}}\left(D_i[s]-A_i[s]\right) \right] \right) \\
	 && \times \text{exp}\left( - \mathbf{1}_{\left\lbrace l \geq 2 \right\rbrace} \sum_{s=1}^{l-1} \left[ \beta_{\text{out} \leftarrow \text{out}}\left(A_i[s+1]-D_i[s]\right) \right] \right) \\
	 && \times \text{exp}\left( - \sum_{t=A_i[l]}^{C_i-1} \left( \sum_{w=1}^{2} \beta_{w_i \leftarrow w} I_w\left(t\right) \right) - \beta_{\text{ward} \leftarrow \text{out}}\left(C_i-A_i[l]\right)  \right) \\
	 && \times \left( 1 - \text{exp}\left( - \sum_{w=1}^{2} \beta_{w_i \leftarrow w} I_w\left(C_i\right) - \beta_{\text{ward} \leftarrow \text{out}} \right) \right) \\
	 && \times \frac{\sum_{j=1}^N \mathbf{1}_{\left\lbrace C_j \leq C_i < E_j \right\rbrace} \sum_{r=1}^{k_j} \mathbf{1}_{\left\lbrace A_j[r] \leq C_i < D_j[r] \right\rbrace} \beta_{w_i \leftarrow w_j} f_{i \leftarrow j} + \beta_{\text{ward} \leftarrow \text{out}}f_{\text{i},\text{ward} \leftarrow \text{out}} }{\sum_{j=1}^N \mathbf{1}_{\left\lbrace C_j \leq C_i < E_j \right\rbrace} \sum_{r=1}^{k_j} \mathbf{1}_{\left\lbrace A_j[r] \leq C_i < D_j[r] \right\rbrace} \beta_{w_i \leftarrow w_j} + \beta_{\text{ward} \leftarrow \text{out}} } \\
\lefteqn{\Omega_i^{\left(3\right)}  = \mathbf{1}_{\left\lbrace k_i > 1 \right\rbrace} \left( 1-\pi \right) \sum_{l=1}^{k_i-1} \mathbf{1}_{\left\lbrace D_i[l] \leq C_i < A_i[l+1] \right\rbrace}}\\
	 && \times \text{exp}\left( - \sum_{s=1}^{l} \left[ \sum_{t=A_i[s]}^{D_i[s]-1} \left( \sum_{w=1}^{2} \beta_{w_i \leftarrow w} I_w\left(t\right) \right) + \beta_{\text{ward} \leftarrow \text{out}}\left(D_i[s]-A_i[s]\right) \right] \right) \\
	 && \times \text{exp}\left( - \mathbf{1}_{\left\lbrace l \geq 2 \right\rbrace} \sum_{s=1}^{l-1} \left[ \beta_{\text{out} \leftarrow \text{out}}\left(A_i[s+1]-D_i[s]\right) \right] \right) \\
	 && \times \text{exp}\left( - \beta_{\text{out} \leftarrow \text{out}}\left(C_i-D_i[l]\right) \right) \\
	 && \times \left( 1 - \text{exp}\left( - \beta_{\text{out} \leftarrow \text{out}} \right) \right) \\
	 && \times f_{\text{i},\text{out} \leftarrow \text{out}}  \\
\end{eqnarray*}
\bigskip

$\Omega_i^{\left(1\right)}$ is the probability that individual $i$ is colonized before his/her first admission in the wards ; $\Omega_i^{\left(2\right)}$ is the probability that individual $i$ is colonized during one of his/her stays in the wards ; $\Omega_i^{\left(3\right)}$ is the probability, that individual $i$, if admitted several times, is colonized between successive stays in the wards.

$f_{i,j}$ is the probability of observing sequences $s_i$ and $s_j$ at respective time steps $t_i$ and $t_j$ given that individual $j$ infected individual $i$ (see next section on the genetic likelihood). Similarly, $f_{\text{i},\text{ward} \leftarrow \text{out}}$ and $f_{\text{i},\text{out} \leftarrow \text{out}}$ are the probability of observing sequence $s_i$ at time $t_i$ given that individual $i$ was infected from outside the wards while he was in or outsite hospital respectively. 

\end{document}

