% Template for PLoS
% Version 1.0 January 2009
%
% To compile to pdf, run:
% latex plos.template
% bibtex plos.template
% latex plos.template
% latex plos.template
% dvipdf plos.template

\documentclass[10pt]{article}

% amsmath package, useful for mathematical formulas
\usepackage{amsmath}
% amssymb package, useful for mathematical symbols
\usepackage{amssymb}

% graphicx package, useful for including eps and pdf graphics
% include graphics with the command \includegraphics
\usepackage{graphicx}

% cite package, to clean up citations in the main text. Do not remove.
\usepackage{cite}

\usepackage{color} 

% Use doublespacing - comment out for single spacing
%\usepackage{setspace} 
%\doublespacing

\usepackage[frenchb,english]{babel}
\usepackage{bm}
\usepackage{cite}
\usepackage{amsthm}
\usepackage{amsfonts}
\usepackage{ccaption}
\usepackage{url}

% Text layout
\topmargin 0.0cm
\oddsidemargin 0.5cm
\evensidemargin 0.5cm
\textwidth 16cm 
\textheight 21cm

% Bold the 'Figure #' in the caption and separate it with a period
% Captions will be left justified
\usepackage[labelfont=bf,labelsep=period,justification=raggedright]{caption}

% Remove brackets from numbering in List of References
\makeatletter
\renewcommand{\@biblabel}[1]{\quad#1.}
\makeatother

\author{In alphabetic order: \\Simon Cauchemez, Anne Cori, Xavier Didelot, \\Neil Ferguson, Christophe Fraser, Thibaut Jombart,\\...}
\title{Reconstructing transmission trees from genetic data: \\a Bayesian approach}

\begin{document}

% Title must be 150 words or less
% \begin{flushleft}
\maketitle

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
\section*{The model, in a nutshell}
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
We seek a simple probabilistic model allowing to reconstruct the transmission tree (who infected whom) of disease outbreaks based on RNA/DNA sequences sampled at given time points.
This model is designed for densely sampled outbreaks of diseases with fairly short (epidemiological) generation times and moderate genetic diversity (typically, genomes should accumulate zero, one or maybe two mutations per generation of infection).
For instance, the method should be relevant for influenza, but HIV is clearly out of the scope of the approach.
\\

The model is inspired by \textit{SeqTrack} in some of the key assumptions it makes: 
\begin{itemize}
\item within-host evolution is considered negligible and mutations only occur during transmission events
\item a single pathogen is considered for each patient (no multi-infection, no within-host diversity)
\item reverse mutations are negligible
\item all cases but the first one trace their ancestry back within the system studied (i.e., no infection from the outside except for the initial case)
\end{itemize}

However, our model aims at improving \textit{SeqTrack} in several respects:
\begin{itemize}
\item a Bayesian framework allowing parameter estimation and incorporating prior information
\item the use of the generation time to compute the likelihood (cf Wallinga \& Teunis)
\item the ability to accommodate unobserved cases
\item the incorporation of infection dates in the transmission model (as augmented data)
\end{itemize}


In a first approach, we assume that the generation time follows a known distribution.
This could be relaxed in a more complex model where parameters of this distribution would be estimated.
The elements we aim to infer are the transmission tree and the mutation rates.




%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
\section*{Data and parameters}
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
\subsection*{Data}
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
For each patient $i=1,\ldots,n$ we note the data:
\begin{itemize}
	\item $s_i$: the genetic sequence obtained for patient $i$.
	\item $t_i$: the collection time for $s_i$ (time is considered as a discrete variable).
%  	\item $d_{i,j}$: the number of transitions between $s_i$ and $s_j$
%  	\item $g_{i,j}$: the number of transversions between $s_i$ and $s_j$
%  	\item $l_{i,j}$: the number of nucleotide positions typed in both $s_i$ and $s_j$
%  	\item $w(\Delta_t)$: likelihood function for a secondary infection occuring $\Delta_t$ unit times after the primary infection; we assume $w(\Delta_t)=0$ for $\Delta_t \leq 0$.
\end{itemize}



%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
\subsection*{Augmented data}
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
Augmented data are noted using capital latin letters:
\begin{itemize}
	\item $T_i^{inf}$: time at which patient $i$ has been infected.
%         \item $\alpha_i$: the closest observed ancestor of $i$ in the infection tree; $\alpha_i=j$ indicates that $j$ has infected $i$, either directly, or with one or several intermediate generations, which were unobserved.
% 	\item $\kappa_i$: an integer $\geq 1$ indicating how many generations separate $\alpha_i$ and $i$: $\kappa_i=1$ indicates that $\alpha_i$ infected $i$; $\kappa_i=2$ indicates that $j$ has infected an unobserved individual, who has in turn infected $i$.
% 	\item $T_i^{ini}$: time at which the most recent observed ancestor of $i$ caused the initial infection of the lineage of $i$. 
% For $\kappa_i=1$, $T_i^{inf} = T_i^{ini}$.
\end{itemize}

% As a first simple approach, $\kappa_i$ could be set to $1$ for all $i$, hence assuming that the whole outbreak was observed.



%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
\subsection*{Functions}
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
We use the following functions of the data/augmented data:
\begin{itemize}
  \item $d(s_i,s_j)$: the number of transitions between $s_i$ and $s_j$.
  \item $g(s_i,s_j)$: the number of transversions between $s_i$ and $s_j$.
  \item $l(s_i,s_j)$: the number of nucleotide positions typed in both $s_i$ and $s_j$.
  \item $w(\Delta_t)$: generation time distribution (likelihood function for a secondary infection occuring $\Delta_t$ unit times after the primary infection); we assume $w(\Delta_t)=0$ for $\Delta_t \leq 0$; while not a requirement in theory, in practice this function will be truncated at a value $\Delta_{max}$ so that $w(\Delta_t)=0$ if $\Delta_t \geq \Delta_{max}$.
  \item $f_w$: a function of the generation time distribution ($w$) indicating how likely it is to sequence an isolate at a given time after infection. By default, we set $f_w=w$, so that the probability of sequencing an isolate is proportional to the infectiousness of the host at the time of collection.
\end{itemize}





%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
\subsection*{Parameters}
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
This model assumes that cases are ordered by increasing infection dates ($T_i^{inf} \leq T_{i+1}^{inf}$).
Parameters are indicated using greek letters:
\begin{itemize}
         \item $\alpha_i$: the closest observed ancestor of $i$ in the infection tree; $\alpha_i=j$ indicates that $j$ has infected $i$, either directly, or with one or several intermediate generations, which were unobserved. 
We note the tree topology $\alpha = \{\alpha_2, \ldots, \alpha_n\}$.
 	\item $\kappa_i$: an integer $\geq 1$ indicating how many generations separate $\alpha_i$ and $i$: $\kappa_i=1$ indicates that $\alpha_i$ infected $i$; $\kappa_i=2$ indicates that $\alpha_i$ has infected an unobserved individual, who has in turn infected $i$.
We note $\kappa = \{\kappa_2, \ldots, \kappa_n\}$.
	\item $\mu_1$: rates of transitions, given per site and per transmission event.
	\item $\mu_2 $: rate of transversions, parametrised as $\mu_2 = \gamma \mu_1$ (with $\gamma \in \mathbb{R}_+$) to account for the correlation between the two rates.
	\item $\pi$: an hyperparameter corresponding to the proportion of observed cases, used to define the priors of $\kappa_i$.
\end{itemize}






%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
\section*{Model}
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
\subsection*{Likelihood}
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%

The posterior distribution is proportional to the joint distribution:
\begin{eqnarray}
& & p(\{s_i, t_i, T_i^{inf}\}_{(i=1,\ldots,n)}, w,  \alpha, \kappa, \mu_1, \gamma, \pi)\\
& = & p(\{s_i, t_i, T_i^{inf}\}_{(i=1,\ldots,n)}| w,  \alpha, \kappa, \mu_1, \gamma, \pi) \times p( w,  \alpha, \kappa, \mu_1, \gamma, \pi)
\end{eqnarray}
where the first term is the likelihood of observed and augmented data, and the second, the prior.
The likelihood can be decomposed as (hyperparameter $\pi$ is dropped as it only affects the likelihood through $\kappa$):

\begin{eqnarray}
& & p(\{s_i, t_i, T_i^{inf}\}_{(i=1,\ldots,n)}|  w,  \alpha, \kappa, \mu_1, \gamma) \\
& = & \prod_{i=2}^n p(s_i, t_i, T_i^{inf} | \{s_k, t_k, T_k^{inf} \}_{(k=1,\ldots,i-1)}, w,  \alpha, \kappa, \mu_1, \gamma) 
  \times p(s_1, t_1, T_1^{inf}|w)\\
& = & \prod_{i=2}^n p(s_i, t_i, T_i^{inf}| s_{\alpha_i}, t_{\alpha_i}, T_{\alpha_i}^{inf},  w,  \alpha_i, \kappa_i, \mu_1, \gamma) 
  \times p(s_1, t_1 , T_1^{inf}|w)\\
& = & \prod_{i=2}^n p(s_i, t_i, T_i^{inf}| s_{\alpha_i}, t_{\alpha_i}, T_{\alpha_i}^{inf},  w,  \alpha_i, \kappa_i, \mu_1, \gamma) 
  \times p(t_1 | T_1^{inf},w) p(T_1^{inf}) p(s_1) 
\end{eqnarray}


$p(t_1 | T_1^{inf},w)$ is the probability of the first collection time given the first infection time. 
The term $p(T_1^{inf}) p(s_1)$ is the probability of the first infection date and of the first sequence, treated as a constant.
This will need to be modified if we explicitely model infections from outside the system.
The term for case $i$ ($i=2,\ldots,n$) is:
\begin{equation}
 p(s_i, t_i, T_i^{inf}| s_{\alpha_i}, t_{\alpha_i}, T_{\alpha_i}^{inf},  w,  \alpha_i, \kappa_i, \mu_1, \gamma )
\end{equation}
which can be decomposed into:
\begin{eqnarray}
& & p(s_i | t_i, T_i^{inf}, s_{\alpha_i}, t_{\alpha_i}, T_{\alpha_i}^{inf},  w,  \alpha_i, \kappa_i, \mu_1, \gamma) \nonumber \\
& &  \times  p(t_i | T_i^{inf}, s_{\alpha_i}, t_{\alpha_i}, T_{\alpha_i}^{inf},  w,  \alpha_i, \kappa_i, \mu_1, \gamma) \nonumber \\
& & \times  p(T_i^{inf}| s_{\alpha_i}, t_{\alpha_i}, T_{\alpha_i}^{inf},  w,  \alpha_i, \kappa_i, \mu_1, \gamma) \nonumber \\
% & & \times  p(\alpha_i, \kappa_i| s_{\alpha_i}, t_{\alpha_i}, T_{\alpha_i}^{inf},  w, \alpha_i, \kappa_i, \mu_1, \gamma)  \\
& = & 
\underbrace{p(s_i | \alpha_i, s_{\alpha_i}, \kappa_i, \mu_1, \gamma)}_{\Omega_i^1} 
  \times  \underbrace{p(t_i | T_i^{inf}, w) 
  p(T_i^{inf}| \alpha_i, T_{\alpha_i}^{inf}, \kappa_i, w)}_{\Omega_i^2}
% & & \times  \underbrace{p(\alpha_i, \kappa_i| s_{\alpha_i}, t_{\alpha_i}, T_{\alpha_i}^{inf},  w, \alpha_i, \kappa_i, \mu_1, \gamma)}_{\Omega_i^3}
% = \underbrace{p(s_i | t_i, T_i^{inf}, \alpha_i, s_{\alpha_i}, t_{\alpha_i}, \mu_1, \gamma)}_{\Omega_i^1}  
%     \underbrace{p(t_i | T_i^{inf}, w)
% 		p(T_i^{inf} | T_i^{ini}, \alpha_i, T_{\alpha_i}^{inf}, \kappa_i, w) % !! ANNE check if T_i^{ini} is relevant here
% 		p(T_i^{ini} | \alpha_i, T_{\alpha_i}^{inf}, \kappa_i, w)}_{\Omega_i^2}
%     \underbrace{p(\alpha_i, \kappa_i | s_{\alpha_i}, t_{\alpha_i}, T_{\alpha_i}^{inf},  w, \alpha, \kappa, \mu_1, \gamma)}_{\Omega_i^3} &
\end{eqnarray}
\noindent where $\Omega_i^1$ is the genetic likelihood and $\Omega_i^2$ if the epidemiological likelihood (derived from W\&T).
% , and $\Omega_i^3$ are priors for the augmented data $\alpha_i$ and $\kappa_i$.
\\




As mutations only occur during transmission events, the expected divergence between two isolates is determined by the number of generations separating these two isolates, and $\Omega_i^1$ is computed as (cf Kimura 1980): 
%% BINOMIAL VERSION %%
\begin{equation}
\underbrace{\mathcal{B}\left(d(s_i,s_{\alpha_i}) | l(s_i,s_{\alpha_i}) \kappa_i, \mu_1 \right)}_{\mbox{transitions}}
\times 
\underbrace{\mathcal{B}\left(g(s_i,s_{\alpha_i}) | l(s_i,s_{\alpha_i}) \kappa_i, \gamma \mu_1 \right)}_{\mbox{transversions}}
\end{equation}
% 
% if $t_{\alpha_i} \leq T_i^{ini} $, and as:
% \begin{equation}
% \underbrace{\mathcal{B}\left(d(i,\alpha_i) | (t_{\alpha_i} - T_i^{ini} + t_i - T_i^{ini}) l(i,\alpha_i), \mu_1 \right)}_{\mbox{transitions}}
% \times 
% \underbrace{\mathcal{B}\left(g(i,\alpha_i) | (t_{\alpha_i} - T_i^{ini} + t_i - T_i^{ini}) l(i,\alpha_i), \gamma \mu_1 \right)}_{\mbox{transversions}}
% \end{equation}
$\mathcal{B}(. | n, p)$ is the probability mass function of a Binomial distribution with $n$ draws and a probability $p$.
This is approximated by:
\begin{equation}
 \underbrace{\mathcal{P}\left(d(s_i,s_{\alpha_i}) | l(s_i,s_{\alpha_i}) \kappa_i \mu_1 \right)}_{\mbox{transitions}}
 \times 
 \underbrace{\mathcal{P}\left(g(s_i,s_{\alpha_i}) | l(s_i,s_{\alpha_i}) \kappa_i \gamma \mu_1 \right)}_{\mbox{transversions}}
 \end{equation}
where $\mathcal{P}(. | \lambda)$ is the density of a Poisson distribution of parameter $\lambda$.
~\\





$\Omega_i^2$ is determined by the distribution of the generation time, and the dates of collection and infection:
\begin{eqnarray}
 \Omega_i^2 & = & p(t_i | T_i^{inf}, w) \times p(T_i^{inf}| \alpha_i, T_{\alpha_i}^{inf}, \kappa_i, w) \nonumber \\
& = &  f_w(t_i - T_i^{inf}) \times  w^{\left(\kappa_i\right)}(T_i^{inf} - T_{\alpha_i}^{inf})
% & = &  \mathbf{1}_{\{w(t_i - T_i^{inf}) > 0\}} \times  w^{\left(\kappa_i\right)}(T_i^{inf} - T_{\alpha_i}^{inf})
\end{eqnarray}
where the first term is the likelihood of the collection date, and the second, the likelihood of the infection time.
$w^{\left(k\right)} = \underbrace{w*w*\ldots*w}_{k \text{ times}} $, where $*$ denotes the convolution operator, defined, for two positive discrete distributions $a$ and $b$, by $\left(a*b\right)\left(t\right) = \sum_{u=0}^{t} a\left(t-u\right)b\left(u\right)$. 
\\

% 
% The term $\Omega_i^3$ can be rewritten as:
% \begin{eqnarray}
% \Omega_i^3 &=&  p(\alpha_i, \kappa_i| s_{\alpha_i}, t_{\alpha_i}, T_{\alpha_i}^{inf},  w, \alpha, \kappa, \mu_1, \gamma) \nonumber \\
% % &=&  p(\alpha_i, \kappa_i,  w, \alpha, \kappa, \mu_1, \gamma)\\
% % 	 &=&  p(w | s_{\alpha_i}, t_{\alpha_i}, T_{\alpha_i}^{inf}) 
% % 	      p(\alpha_i, \kappa_i,\mu_1, \gamma | s_{\alpha_i}, t_{\alpha_i}, T_{\alpha_i}^{inf}) \\
% % 	 &=&  p(w | s_{\alpha_i}, t_{\alpha_i}, T_{\alpha_i}^{inf}) \nonumber \\
% %       & &     p(\alpha_i | s_{\alpha_i}, t_{\alpha_i}, T_{\alpha_i}^{inf})\nonumber  \\
% %       & &     p(\kappa_i | s_{\alpha_i}, t_{\alpha_i}, T_{\alpha_i}^{inf}) \nonumber \\
% %       & &     p(\mu_1 | s_{\alpha_i}, t_{\alpha_i}, T_{\alpha_i}^{inf})\nonumber  \\
% %       & &     p(\gamma | s_{\alpha_i}, t_{\alpha_i}, T_{\alpha_i}^{inf})\\
%       & = &   p(\alpha_i) p(\kappa_i)
% \end{eqnarray}
% as the different components are independent.
% % $p(w)$ is a constant and does not need to be known to sample from (1).
% $p(\alpha_i)$ is the prior on ancestries, set to $1/(n-1)$.
% $p(\kappa_i)$ is the prior on the number of unobserved transmission steps.
% This is given by a binomial distribution of parameter $\pi$, which is the proportion of observed (sampled) cases in the outbreak.
% If we assume that the entire outbreak has been sampled, this would be set to $p(\kappa_i) = \mathbf{1}_{\left\lbrace \kappa_i=1\right\rbrace}$.
% Alternatively, more flexibility would be gained by using a Poisson distribution to allow for unobserved intermediate cases.
% , where $\mathbf{1}_{\left\lbrace.\right\rbrace}$ denotes the indicator function, defined by $\mathbf{1}_{\left\lbrace X \right\rbrace}=1$ if $X$ is true, and $0$ otherwise. 
% $ p(\mu_1)$ and $p(\gamma)$ are the priors for these two parameters.
~\\


% 
% 
% %%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
% %%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
% \section*{Model with missing data}
% %%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
% %%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
% !!! Need to work on that one. 
% 
% The basic model can be extended to incorporate the fact that some infectors have not been sampled, and therefore some infections cannot be reconstructed.
% % This will be modeled by $\alpha_i=0$ when $i$'s infector has not been observed. 
% We introduce the parameter $\pi$, which is the probability of having observed the infector of a given case.
% Then the expression (1) becomes:
% \begin{equation}
%  p(s_i, t_i, T_i^{inf}, \alpha_i,  w, \alpha, \kappa, \mu_1, \gamma, \pi)
% \end{equation}
% which can be decomposed in:
% \begin{eqnarray}
% & & p(s_i | t_i, T_i^{inf}, \alpha_i,  w, \alpha, \kappa, \mu_1, \gamma, \pi)  p(t_i, T_i^{inf}, \alpha_i,  w, \alpha, \kappa, \mu_1, \gamma, \pi)\\
% &=& p(s_i | t_i, T_i^{inf}, \alpha_i,  w, \alpha, \kappa, \mu_1, \gamma, \pi)  p(T_i^{inf} | t_i, \alpha_i,  w, \alpha, \kappa, \mu_1, \gamma, \pi) p(t_i, \alpha_i,  w, \alpha, \kappa, \mu_1, \gamma, \pi)\\
% &=& \underbrace{p(s_i | t_i, T_i^{inf}, \alpha_i, \mu_1, \gamma, \pi)}_{\Omega^*_1}  
%     \underbrace{p(T_i^{inf} | \alpha_i, w, \pi)}_{\Omega^*_2}
%     \underbrace{p(t_i, \alpha_i,  w, \alpha, \kappa, \mu_1, \gamma, \pi)}_{\Omega^*_3} 
% \end{eqnarray}
% The first two terms are simply given by $\Omega^*_1 = \pi \Omega_i^1$ and $\Omega^*_2 = \pi \Omega_i^2$
% 
% % 
% 
% 
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
\subsection*{Priors}
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
%%%%%%%

For all model parameters, independent prior distributions have been chosen:
\begin{itemize}
	\item $p(w) = \mathbf{1}_{\{w=w_0\}}$: the distribution of the generation time will be fixed to a given distribution ($w_0$) by default; this can be parameterized later in a more complex model.
	\item $p(\alpha_i) = \mathbf{1}_{\{k\neq i\}}\frac{1}{n-1}$.
	\item $p(\kappa_i - 1) = \mathcal{NB}(1,1-\pi)$, the probability mass function of a negative binomial distribution counting the number of unobserved cases before one observed case; $1-\pi$ is the proportion of unobserved (unsampled) cases in the outbreak. If we assume that the entire outbreak has been sampled, $\pi=1$ and $p(\kappa_i) = \mathbf{1}_{\left\lbrace \kappa_i=1\right\rbrace}$. Otherwise, $\pi$ will be assumed to follow a Beta distribution of fixed parameters (e.g. $p(\pi) = \beta(4,3)$).
%       \kappa, \mu_1, \gamma uniform
	\item $p(\mu_1) = Unif(0,1)$.
	\item $p(\gamma) = log\mathcal{N}(1,1.25)$ where $log\mathcal{N}(\mu,\sigma)$ is the log-normal distribution with mean $\mu$ and standard deviation $\sigma$ (values are the default in BEAST).
% 	\item $p(\mu_1) = Exp(\lambda1)$, an exponential distribution of parameter $\lambda1$.
% 	\item $p(\gamma)$
\end{itemize}


% $p(\alpha_i)$ is the prior on ancestries, set to $1/(n-1)$.
% $p(\kappa_i)$ is the prior on the number of unobserved transmission steps.
% This is given by a binomial distribution of parameter $\pi$, which is the proportion of observed (sampled) cases in the outbreak.
% If we assume that the entire outbreak has been sampled, this would be set to $p(\kappa_i) = \mathbf{1}_{\left\lbrace \kappa_i=1\right\rbrace}$.




%   
% \section*{Parameter Estimation}
% 
% A Markov chain Monte Carlo (MCMC) method was used to sample the joint
% posterior distribution $P\left(\bm{Y},\bm{Z},\bm{\theta}\right)$.
% $\psi$, $\phi$ and $\pi$ were updated using the Gibbs sampler, and all other parameters using a Metropolis algorithm.



\end{document}

